python_modules/dagster/dagster/_daemon/backfill.py

```diff
@@ -38,10 +40,14 @@ def execute_backfill_iteration(
                 yield from execute_job_backfill_iteration(
                     backfill, logger, workspace_process_context, debug_crash_flags, instance
                 )
-        except Exception:
+        except Exception as e:
             error_info = serializable_error_info_from_exc_info(sys.exc_info())
-            instance.update_backfill(
-                backfill.with_status(BulkActionStatus.FAILED).with_error(error_info)
-            )
-            logger.error(f"Backfill failed for {backfill.backfill_id}: {error_info.to_string()}")
+            if os.environ.get("RETRY_CODE_LOCATION_ERROR") and isinstance(e, DagsterAssetBackfillDataLoadError):
+                logger.error(f"Backfill error for {backfill.backfill_id}: {error_info.to_string()}. Will retry.")
```

Comment:
I don't think we need to now, but in the future we could update the error here and make the ui surface an error even while the backfill isn't failed